packages: containerd 1.4.1

NicolasT opened the pull request:
Changes, including the patch, based on `containerd-1.4.1-1.fc34`. Still
using the RHEL7/EPEL way of packaging Go projects.
 
Closes: #2855
See: https://github.com/scality/metalk8s/issues/2855

### TODO
- [x] Changelog
- [ ] Should we set `Vendor` and/or a particular string in `Release`?

NicolasT committed:
buildchain: teach `package` target about `Patch` sources

Next to `SourceN` stanzas, RPM also has `PatchN` lines, which are, in a
sense, sources of the package. This patch (pun intended) teaches the
target to properly parse these lines.

NicolasT committed:
packages, scripts: containerd 1.4.1

Changes, including the patch, based on `containerd-1.4.1-1.fc34`. Still
using the RHEL7/EPEL way of packaging Go projects.

In the `backup.sh` script, properly handle the 'v2' runtime to retrieve
the etcd snapshot database from `containerd`'s `/run/containerd/...`
directory structure.
Note: we support *both* structures, since after an upgrade, the 'v2'
structure is *not* used unless the container gets recreated, and backups
should remain possible at all times.

Closes: #2855
See: https://github.com/scality/metalk8s/issues/2855

bert-e commented:
# Hello nicolast,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @nicolast

* @lucieleonard

* @alexandre-allard-scality

* @monpote

* @slaperche-scality

* @teddyandrieux

* @cmonfort

* @thomasdanan

* @carlito-scality

* @jbertran

* @sayf-eddine-scality

* @gdemonet

* @nootal

NicolasT committed:
buildchain: no `noexec` on `/home/build`

When building an RPM package, we mount a `tmpfs` filesystem on
`/home/build`, for which Docker uses the `noexec` mount flag by default.
In general, this works fine, until some RPM package requires to
*execute* a binary built as part of the build process to generate some
files. Looking at you, `containerd` `gen-manpages`.

To support such builds, mount the `/home/build` `tmpfs` using the `exec`
flag instead.

See: https://github.com/scality/metalk8s/pull/2869

NicolasT committed:
scripts: add some assertions to etcd backup code

Starting with `containerd` 1.4.1, the 'v2' runtime is used. This changes
the paths used under `/run/containerd`. This patch prepares to support
these various options, and adds a couple of assertions to bail out when
certain expectations are not met.

NicolasT committed:
scripts: delete etcd snapshot from container rootfs

Once we copied the etcd snapshot file into the backup directory, we
should remove it from the container rootfs to reclaim storage.

NicolasT committed:
Merge remote-tracking branch 'origin/development/2.7' into improvement/containerd-1.4.1

* origin/development/2.7:
  packages: bump `containerd` to 1.2.14
  packages: bump `containerd` to 1.2.13
  build: build 'our' `containerd` RPM
  packages: import SRPM files of `containerd-1.2.4-1.el7`
  build: Bump K8s version to 1.18.10
  script: Backup before and after upgrade and downgrade
  build: Bump K8s verison to 1.17.13
  salt: no longer user pillar to get available solutions

Conflicts:
	buildchain/buildchain/versions.py
	packages/redhat/containerd.spec

NicolasT commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.7`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.7`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`



Please check the status of the associated issue None.

Goodbye nicolast.
